frontend/packages/component-library/src/tooltip/WithTooltip.tsx

```diff
@@ -85,12 +100,31 @@ const WithTooltip: React.FunctionComponent<WithTooltipProps> = ({
     updateTooltipStyles,
   ]);
 
+  // Effect to handle the Escape key to close the tooltip
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !event.defaultPrevented) {
+        if (event.target instanceof HTMLElement) {
+          event.target.blur(); // Remove focus from the container
+        } else {
+          handleHideTooltip(); // Hide the tooltip if unable to remove focus natively
+        }
+      }
+    };
+    if (showTooltip) {
+      window.addEventListener('keydown', handleKeyDown);
+    }
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [showTooltip]);
+
   const tooltipStyleProps: React.CSSProperties = {
     visibility: showTooltip ? 'visible' : 'hidden',
     ...tooltipStyles,
   };
 
-  // Check if children is a valid React element and clone it with ariaDescribedBy attribute
+  // Check if child is a valid React element and clone it with ariaDescribedBy attribute
```

Comment:
nit: if choose between two, `children prop` is the most accurate here I think
```suggestion
  // Check if children prop is a valid React element and clone it with ariaDescribedBy attribute
```